packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -438,9 +450,40 @@ export function getDevMiniflareOptions(
 							),
 						},
 					],
+					unsafeUseModuleFallbackService: true,
 				} satisfies WorkerOptions;
 			}),
 		],
+		unsafeModuleFallbackService(request) {
+			const url = new URL(request.url);
+			const rawSpecifier = url.searchParams.get("rawSpecifier");
+			assert(
+				rawSpecifier,
+				`Unexpected error: no specifier in request to module fallback service.`
+			);
+
+			const moduleRE = new RegExp(MODULE_PATTERN);
+			const match = moduleRE.exec(rawSpecifier);
+			assert(match, `Unexpected error: no match for module ${rawSpecifier}.`);
+			const [full, moduleType, modulePath] = match;
+			assert(
+				modulePath,
+				`Unexpected error: module path not found in reference ${full}.`
+			);
+
+			let source: Buffer;
+
+			try {
+				source = fs.readFileSync(modulePath);
+			} catch (error) {
+				throw new Error(`Import ${modulePath} not found. Does the file exist?`);
+			}
+
+			return MiniflareResponse.json({
+				// Cap'n Proto expects byte arrays for `:Data` typed fields from JSON
+				wasm: source instanceof Uint8Array ? Array.from(source) : source,
```

Comment:
Isn't `source` always a `Buffer`? In which case it is also always a `Uint8Array`?

Reply:
Yes. I wasn't sure if this would be the case when adding the other module types so put it here to remind me. Copied from https://github.com/cloudflare/workers-sdk/blob/0c0374cce3908a47f7459ba4810855c1ce124349/packages/vitest-pool-workers/src/pool/module-fallback.ts#L432.

Reply:
[d4c4077](https://github.com/cloudflare/workers-sdk/pull/8031/commits/d4c4077d4107d02456af625fa674202142d1764f)